tools/converter/binary_reader/ReadManager.h

```diff
@@ -0,0 +1,723 @@
+        // these typedefs define from which STL some of the clases are derived from
+
+        typedef std::set<bbpReader::CellInfo> FrameInfo;
+        typedef std::vector<MappingItem> CompartmentMapping;
+        typedef std::vector<DataItem> SectionData;
+        // typedef std::map<bbpReader::SegmentData::Identifier,bbpReader::SegmentData>
+        // CellData; done blow
+        typedef std::map<bbpReader::CellInfo, bbpReader::CellData> FrameOrganiser;
+        typedef std::vector<std::ios::off_type> ReadOptimiser;
+    }
+
+    // end of type conventions
+
+    class Header {
+      private:
+        bool isNative;
+
+        // header info
+        uint32_t headerSize;
+        std::string libraryVersion;
+        std::string simulatorVersion;
+        CellCount amountOfCells;
+        FrameDataCount totalAmountOfCompartments;
+        FrameCount numberOfSteps;
+        Time startTime;
+        Time endTime;
+        Time dt;
+        std::string dataUnit;
+        std::string timeUnit;
+        FrameDataCount mappingSize;
+        std::string mappingName;
+        FrameDataCount extraMappingSize;
+        std::string extraMappingName;
+        std::string targetName;
+
+      public:
+        Header() : headerSize(0){}
+        Header(char* readBuffer);
+        // Header(const Header& source);
+
+        bool operator==(const Header& h1) const;
+
+        // get methods:
+        inline bool sourceIsNative() const {
+            return isNative;
+        }
+        inline std::string getLibraryVersion() const {
+            return libraryVersion;
+        }
+        inline std::string getSimulatorVersion() const {
+            return simulatorVersion;
+        }
+        inline CellCount getNumberOfCells() const {
+            return amountOfCells;
+        }
+        inline FrameDataCount getTotalNumberOfCompartments() const {
+            return totalAmountOfCompartments;
+        }
+        inline FrameCount getNumberOfSteps() const {
+            return numberOfSteps;
+        }
+        inline std::string getMappingName() const {
+            return mappingName;
+        }
+        inline std::string getExtraMappingName() const {
+            return extraMappingName;
+        }
+        inline std::string getDataUnit() const {
+            return dataUnit;
+        }
+        inline std::string getTimeUnit() const {
+            return timeUnit;
+        }
+        inline FrameDataCount getMappingSize() const {
+            return mappingSize;
+        }
+        inline FrameDataCount getExtraMappingSize() const {
+            return extraMappingSize;
+        }
+        inline std::string getTargetName() const {
+            return targetName;
+        }
+        inline Time getStartTime() const {
+            return startTime;
+        }
+        inline Time getEndTime() const {
+            return endTime;
+        }
+        inline Time getTimeStepSize() const {
+            return dt;
+        }
+        // end get methods
```

Comment:
`nitpick`: I understand that this is old code but seeing inconsistent naming (camelcase vs snake case) is a bit not "ideal".

Reply:
same :(